File: stack/682.baseball-game.go
package main

import "strconv"

/*
 * @lc app=leetcode id=682 lang=golang
 *
 * [682] Baseball Game
 */

// @lc code=start
func calPoints(ops []string) int {
	result := []int{}
	for _, op := range ops {
		switch op {
		case "C":
			result = result[:len(result)-1]
		case "D":
			result = append(result, result[len(result)-1]*2)

		case "+":
			result = append(result, result[len(result)-1]+result[len(result)-2])
		default:
			num, _ := strconv.Atoi(op)
			result = append(result, num)
		}
	}
	return sum(result)
}

func sum(list []int) int {
	s := 0
	for _, v := range list {
		s += v
	}
	return s
}

// @lc code=end

File: stack/682.baseball-game.py
from typing import List

#
# @lc app=leetcode id=682 lang=python3
#
# [682] Baseball Game
#

# @lc code=start


class Solution:
    def calPoints(self, ops: List[str]) -> int:
        result: List[int] = []
        for s in ops:
            if s == "C":
                result.pop()
            elif s == "+":
                s1 = result.pop()
                s2 = result.pop()
                result.append(s2)
                result.append(s1)
                result.append(s1 + s2)
            elif s == "D":
                s1 = result.pop()
                result.append(s1)
                result.append(s1 * 2)
            else:
                result.append(int(s))
        return sum(result)


# @lc code=end

if __name__ == "__main__":
    s = Solution()
    ops = ["5", "-2", "4", "C", "D", "9", "+", "+"]

    print(s.calPoints(ops))
